composable fsdp, checkpoint, + compile test

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #106003
* #106002
* #106001
* #106000
* #105999
* #105995
* #105991
* #105207
* #105206
* #105205
* __->__ #105180
* #105176

Test to ensure that composable FSDP, checkpoint, and compile all work
together. Includes a change from https://github.com/pytorch/pytorch/pull/105090
which we can land in that PR first.

Differential Revision: [D47452973](https://our.internmc.facebook.com/intern/diff/D47452973/)

rohan-varma committed:
composable fsdp, checkpoint, + compile test

Test to ensure that composable FSDP, checkpoint, and compile all work
together. Includes a change from https://github.com/pytorch/pytorch/pull/105090
which we can land in that PR first.

Differential Revision: [D47452973](https://our.internmc.facebook.com/intern/diff/D47452973/)

[ghstack-poisoned]

rohan-varma committed:
Update on "composable fsdp, checkpoint, + compile test"


Test to ensure that composable FSDP, checkpoint, and compile all work
together. Includes a change from https://github.com/pytorch/pytorch/pull/105090
which we can land in that PR first.

Differential Revision: [D47452973](https://our.internmc.facebook.com/intern/diff/D47452973/)

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/_runtime_utils.py:
```diff
@@ -443,6 +443,18 @@ def _pre_forward(
         kwargs (Dict[str, Any]): Module forward ``kwargs``.
     """
     with torch.profiler.record_function("FullyShardedDataParallel._pre_forward"):
+        # Land as part of https://github.com/pytorch/pytorch/pull/105090
```
We can remove this since our plan is to not land that one first now.
```suggestion
```

awgu commented on test/distributed/_composable/fully_shard/test_fully_shard_compile.py:
```diff
@@ -41,12 +41,13 @@ def test_compile(self):
                     ShardingStrategy.NO_SHARD,
                 ],
                 "skip_fsdp_guards": [True, False],
+                "act_checkpoint": [True],
```
Should we still include `False`?

rohan-varma committed:
Update on "composable fsdp, checkpoint, + compile test"


Test to ensure that composable FSDP, checkpoint, and compile all work
together. Includes a change from https://github.com/pytorch/pytorch/pull/105090
which we can land in that PR first.

Differential Revision: [D47452973](https://our.internmc.facebook.com/intern/diff/D47452973/)

[ghstack-poisoned]

awgu reviewed (approved):
Stamp to unblock -- should we patch over `_post_forward` change as well?

rohan-varma committed:
Update on "composable fsdp, checkpoint, + compile test"


Test to ensure that composable FSDP, checkpoint, and compile all work
together. Includes a change from https://github.com/pytorch/pytorch/pull/105090
which we can land in that PR first.

Differential Revision: [D47452973](https://our.internmc.facebook.com/intern/diff/D47452973/)

[ghstack-poisoned]

rohan-varma committed:
Update on "composable fsdp, checkpoint, + compile test"


Test to ensure that composable FSDP, checkpoint, and compile all work
together. Includes a change from https://github.com/pytorch/pytorch/pull/105090
which we can land in that PR first.

Differential Revision: [D47452973](https://our.internmc.facebook.com/intern/diff/D47452973/)

[ghstack-poisoned]

rohan-varma committed:
Update on "composable fsdp, checkpoint, + compile test"


Test to ensure that composable FSDP, checkpoint, and compile all work
together. Includes a change from https://github.com/pytorch/pytorch/pull/105090
which we can land in that PR first.

Differential Revision: [D47452973](https://our.internmc.facebook.com/intern/diff/D47452973/)

[ghstack-poisoned]
